ndb_api/auth.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ndb_api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Validates the auth credentials against the 'auth/validate' endpoint
+// Returns the response along with an error (if any)
+func AuthValidate(ctx context.Context, ndbClient *ndb_client.NDBClient) (authValidateResponse AuthValidateResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered ndb_api.AuthValidate")
+	if ndbClient == nil {
+		err = errors.New("nil reference: received nil reference for ndbClient")
+		log.Error(err, "Received nil ndbClient reference")
+		return
+	}
+	res, err := ndbClient.Get("auth/validate")
```

Comment:
nit; we should consolidate all these hardcoded API paths into the constants file. We need to update other places as well when we fix that.

Reply:
Cool suggestion! I'll include this in the ndb_api refactoring changes we discussed above!